pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,145 @@ type ExternalNodeList struct {
 
 	Items []ExternalNode `json:"items,omitempty"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleCollectionList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []SupportBundleCollection `json:"items"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type SupportBundleCollection struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Specification of the desired behavior of SupportBundleCollection.
+	Spec SupportBundleCollectionSpec `json:"spec"`
+	// Most recently observed status of the SupportBundleCollection.
+	Status SupportBundleCollectionStatus `json:"status"`
+}
+
+type SupportBundleCollectionSpec struct {
+	Nodes         *BundleNodes         `json:"nodes,omitempty"`
+	ExternalNodes *BundleExternalNodes `json:"externalNodes,omitempty"`
+	// ExpirationMinutes is the requested duration of validity of the SupportBundleCollection.
+	// A SupportBundleCollection will be marked as Failed if it does not finish before expiration.
+	// Default is 60.
+	ExpirationMinutes int32 `json:"expirationMinutes"`
+	// SinceTime specifies a relative time before the current time from which to collect logs
+	// A valid value is like: 1d, 2h, 30m.
+	SinceTime      string                        `json:"sinceTime,omitempty"`
+	FileServer     BundleFileServer              `json:"fileServer"`
+	Authentication BundleServerAuthConfiguration `json:"authentication"`
+}
+
+// BundlePhase defines the phase in which a SupportBundleCollection is.
+type BundlePhase string
+
+const (
+	// BundlePending means the SupportBundleCollection has been accepted by the system,
+	// but it has not been processed by Antrea.
+	BundlePending BundlePhase = "Pending"
+	// BundleProcessing means the SupportBundleCollection is being processed by Antrea.
+	BundleProcessing BundlePhase = "Processing"
+	// BundleSucceeded means the SupportBundleCollection is successfully processed by all
+	// required Nodes or ExternalNodes.
+	BundleSucceeded BundlePhase = "Succeeded"
+	// BundlePartiallySucceeded means some required Nodes or ExternalNodes have successfully
+	// processed the SupportBundleCollection request, but others failed.
+	BundlePartiallySucceeded BundlePhase = "PartiallySucceeded"
+	// BundleFailed means the SupportBundleCollection failed. It can happen when Antrea
+	// Controller fails to create internal objects, or all of the required Nodes or
+	// ExternalNodes fail to process the request.
+	BundleFailed BundlePhase = "Failed"
+)
+
+type SupportBundleCollectionStatus struct {
+	Phase BundlePhase `json:"phase"`
+	// The generation observed by Antrea.
+	ObservedGeneration int64 `json:"observedGeneration"`
```

Comment:
Do we support updating bundle spec? If not, this field will be pointless.

Reply:
Removed this field.